src/DoctrineCheckpointStorage.php

```diff
@@ -36,16 +36,16 @@ public function acquireLock(): SequenceNumber
                 'subscriberId' => $this->subscriberId
             ]);
         } catch (DriverException $exception) {
+            $this->connection->rollBack();
             // Error code "1205" = ER_LOCK_WAIT_TIMEOUT in MySQL (https://dev.mysql.com/doc/refman/8.0/en/server-error-reference.html#error_er_lock_wait_timeout)
             // SQL State "55P03" = lock_not_available in PostgreSQL (https://www.postgresql.org/docs/9.4/errcodes-appendix.html)
             if ($exception->getErrorCode() === 1205 || $exception->getSQLState() === '55P03') {
                 throw new CheckpointException(sprintf('Failed to acquire checkpoint lock for subscriber "%s" because it is acquired already', $this->subscriberId), 1652279016);
             }
             throw new \RuntimeException($exception->getMessage(), 1544207633, $exception);
         } catch (DBALException $exception) {
-            throw new \RuntimeException($exception->getMessage(), 1544207778, $exception);
-        } finally {
             $this->connection->rollBack();
```

Comment:
transaction rollback in finally() won't work, as it will release the FOR UPDATE lock. This essentially defeats the purpose of the DoctrineCheckpointStorage :)